Created getting started wtih gitops procedure module

jc-berger opened the pull request:
- Adds an overview for GitOps and addresses Jira Issue, #2475.

- Targeted to OCP 4.7

- Aligned team label: Dev Tools

- Acknowledged by QE. See Shoubhik's "looks good to me" comment. 

- For easy reference, here's a preview of the [rendered version](https://deploy-preview-29411--osdocs.netlify.app/openshift-enterprise/latest/cicd/gitops/getting-started-with-openshift-gitops.html#configuring-argo-cd-openid-connect-oidc ).

Preeticp commented on modules/getting-started-openshift-gitops.adoc:
```diff
@@ -0,0 +1,128 @@
+// * openshift-docs/cicd/gitops/working-with-openshift-gitops.adoc
+
+[id="getting-started-with-red-hat-openshift-gitops_{context}"]
+= Getting Started with Red Hat OpenShift GitOps
```
This is a collection of procedure modules, we will need to create an assembly for this and add the different modules to it. We will also need to add an entry to the Topic map file.

Preeticp commented on modules/getting-started-openshift-gitops.adoc:
```diff
@@ -0,0 +1,128 @@
+[id="installing-red-hat-openshift-gitops_{context}"]
+== Installing Red Hat OpenShift GitOps
+
+.Prerequisite
```
Please use a bulleted list for Prerequisites, see here: https://docs.openshift.com/container-platform/4.6/pipelines/creating-applications-with-cicd-pipelines.html#prerequisites

jc-berger replied:
Add the prereq with links into the assembly. that way, you're not adding any links into the module.

Preeticp commented on modules/getting-started-openshift-gitops.adoc:
```diff
@@ -0,0 +1,128 @@
+
+.Prerequisite
+
+The Red Hat product version of Argo CD is the Openshift GitOps Operator. When you install the Red Hat OpenShift GitOps operator, you get an instance of Argo CD in the 'openshift-gitops' namespace.
```
Remove the sentence: The Red Hat product version of Argo CD is the Openshift GitOps Operator.
Suggest something like: 
* You have installed the Red Hat OpenShift GitOps operator listed in the OpenShift OperatorHub. Once installed, it is applicable to the entire cluster. When you install the Red Hat OpenShift GitOps operator, you get an instance of Argo CD in the 'openshift-gitops' namespace.

Preeticp replied:
Another reason why we need an assembly, no xrefs are allowed from modules, so prereqs with links are added to the assembly.

jc-berger replied:
Add sentence at the end about how it also adds an Argo CD icon  in the console toolbar.

jc-berger replied:
Might have to mention the precise areas where it's shown.

Preeticp commented on modules/getting-started-openshift-gitops.adoc:
```diff
@@ -0,0 +1,128 @@
+
+The Red Hat product version of Argo CD is the Openshift GitOps Operator. When you install the Red Hat OpenShift GitOps operator, you get an instance of Argo CD in the 'openshift-gitops' namespace.
+
+To ensure you can work with the Red Hat OpenShift GitOps operator, go to xref:../operators/admin/olm-adding-operators-to-cluster.adoc#adding-operators-to-a-cluster[installing an operator from OperatorHub].
```
Remove this

Preeticp commented on modules/getting-started-openshift-gitops.adoc:
```diff
@@ -0,0 +1,128 @@
+
+To ensure you can work with the Red Hat OpenShift GitOps operator, go to xref:../operators/admin/olm-adding-operators-to-cluster.adoc#adding-operators-to-a-cluster[installing an operator from OperatorHub].
+
+After you install an Operator, you have a Red Hat OpenShift Container cluster. Continue with the default operator installation because this operator installs to all namespaces on the cluster. Now you have installed a Red Hat OpenShift GitOps operator.
```
Remove this

Preeticp commented on modules/getting-started-openshift-gitops.adoc:
```diff
@@ -0,0 +1,128 @@
+
+The Red Hat OpenShift GitOps operator automatically sets up a ArgoCD cluster configuration. ArgoCD manages specific cluster-scoped resources, including: platform operators, optional Operator Lifecycle Manager (OLM) operators, and user management. Because ArgoCD is not a cluster-admin, you must launch into the ArgoCD instance from the Console Application Launcher.
+
+To find the password for the ArgoCD:
```
Add this as the second step: Enter the password to log into the ArgoCD UI.
And then say - To find the password for the ArgoCD:
Steps on line 26 to 31 should all be nested steps under the above step.

Preeticp commented on modules/getting-started-openshift-gitops.adoc:
```diff
@@ -0,0 +1,128 @@
+
+To find the password for the ArgoCD:
+
+. Go to the table of contents and make sure you are in the *Developer* perspective and not the *Administrator* perspective.
```
remove this step, add it to the first step as suggested above.

Preeticp commented on modules/getting-started-openshift-gitops.adoc:
```diff
@@ -0,0 +1,128 @@
+. Copy the password.
+. Use this password to log into the ArgoCD UI.
+
+*Note:* if you want to fetch the ArgoCD password from the command line interface, run the following command:
```
This note must be added below the nested steps.

See OpenShift docs guidelines for the correct format to apply for admonitions like Notes

s/if you want to fetch/ To fetch

Preeticp commented on modules/getting-started-openshift-gitops.adoc:
```diff
@@ -0,0 +1,128 @@
+
+*Note:* if you want to fetch the ArgoCD password from the command line interface, run the following command:
+
+ `oc get secret argocd-cluster-cluster -n openshift-gitops -o jsonpath='{.data.admin\.password}' | base64 -d`
```
Format for command lines is provided in the OCP docs repo, doc guidelines. For example, see: https://raw.githubusercontent.com/openshift/openshift-docs/master/modules/op-assembling-a-pipeline.adoc
For your case a simple [source, terminal] would probably do.
Add $ before the command.

Preeticp commented on modules/getting-started-openshift-gitops.adoc:
```diff
@@ -0,0 +1,128 @@
+
+[id="setting-up-a-new-argocd-instance_{context}"]
+== Setting up a new ArgoCD instance
+
```
`openshift-gitops` namespace

Preeticp commented on modules/getting-started-openshift-gitops.adoc:
```diff
@@ -0,0 +1,128 @@
+
+To install an ArgoCD instance:
+
+. Go to the table of contents and make sure you are in the *Administrator* perspective and not the *Developer* perspective.
```
In the *Administrator* perspective of the web console, navigate to *Operators* → *Installed Operators*.
This should be good for both the steps on line 48 and 49

Preeticp commented on modules/getting-started-openshift-gitops.adoc:
```diff
@@ -0,0 +1,128 @@
+. Go to the table of contents and make sure you are in the *Administrator* perspective and not the *Developer* perspective.
+. Select *Operators* then select *Installed Operators*.
+. Create a new project or select an existing project, depending on where you want to install the ArgoCD instance.
+. Select *OpenShift GitOps* from the "Installed Operators" page.
```
All GUI elements in bold (single asterix in asciidoc. The comments here use markdown so will render the single asterix into italics). Location, then action: In the *Installed Operators* page, select *OpenShift GitOps*

Preeticp commented on modules/getting-started-openshift-gitops.adoc:
```diff
@@ -0,0 +1,128 @@
+. Create a new project or select an existing project, depending on where you want to install the ArgoCD instance.
+. Select *OpenShift GitOps* from the "Installed Operators" page.
+. Select the *ArgoCD* tab.
+. Create an ArgoCD instance and specify the following configurations:
```
Which GUI action do you need to perform to create an ArgoCD instance?

jc-berger replied:
Not sure. I'll have to look more into that!

Preeticp commented on modules/getting-started-openshift-gitops.adoc:
```diff
@@ -0,0 +1,128 @@
+
+.Deploying resources to a different namespace
+
+To grant ArgoCD the permissions to manage resources in multiple namespaces, create RoleBindings which provide ArgoCD admin permissions for that particular namespace. Use the example, OOTB ArgoCD instance, deployed in the openshift-gitops namespace.
```
RoleBindings/role bindings,

`OOTB ArgoCD` instance, deployed in the `openshift-gitops` namespace.

Preeticp commented on modules/getting-started-openshift-gitops.adoc:
```diff
@@ -0,0 +1,128 @@
+To create a RoleBinding:
+
+. Go to the table of contents and make sure you are in the *Administrator* perspective and not the *Developer* perspective.
+. Go to *User Management* > *Role Bindings* > select *Create Binding*.
```
Modify the above steps as per this style: In the *xyz* perspective of the web console, navigate to *abc* -> *def*.

Preeticp commented on modules/getting-started-openshift-gitops.adoc:
```diff
@@ -0,0 +1,128 @@
+
+. Go to the table of contents and make sure you are in the *Administrator* perspective and not the *Developer* perspective.
+. Go to *User Management* > *Role Bindings* > select *Create Binding*.
+  * You now see a form that guides you through the process of creating a RoleBinding:
```
Modify the above steps on line 63, 64, 65 as per this style: In the xyz perspective of the web console, navigate to abc -> def to see the role binding creation form(replace with the exact name of the form in the GUI).
Please test this on a cluster or see the recording to see what this form is called and replace it above

jc-berger replied:
Will do tomorrow. It's so late there's no longer a fresh cluster :) When I can't check it on a cluster myself, I rely on the screenshots in our shared Gdoc.

Preeticp commented on modules/getting-started-openshift-gitops.adoc:
```diff
@@ -0,0 +1,128 @@
+. Go to *User Management* > *Role Bindings* > select *Create Binding*.
+  * You now see a form that guides you through the process of creating a RoleBinding:
+. For *Binding Type*, select *Namespace Role Binding (RoleBinding)*.
+  * If you only select *All Projects*, then you only need to provide the *Binding Type*.
```
Remove bullet style for the above sentence.

Preeticp commented on modules/getting-started-openshift-gitops.adoc:
```diff
@@ -0,0 +1,128 @@
+  * You now see a form that guides you through the process of creating a RoleBinding:
+. For *Binding Type*, select *Namespace Role Binding (RoleBinding)*.
+  * If you only select *All Projects*, then you only need to provide the *Binding Type*.
+  * If the project is already selected as the one you want to create the RoleBinding for, the project does not ask for *Binding Type* because it assumes it is a RoleBinding.
```
Probably combine content on line 67 and 68 something like this: 
You need to provide the *Binding Type* only when you select *All Projects*.  You do not need to specify the binding type if a particular project is selected, as the type is assumed to be role binding for the project.

Preeticp commented on modules/getting-started-openshift-gitops.adoc:
```diff
@@ -0,0 +1,128 @@
+. For *Binding Type*, select *Namespace Role Binding (RoleBinding)*.
+  * If you only select *All Projects*, then you only need to provide the *Binding Type*.
+  * If the project is already selected as the one you want to create the RoleBinding for, the project does not ask for *Binding Type* because it assumes it is a RoleBinding.
+. Under *Role Binding*, provide a RoleBinding *Name* and *Namespace*.
```
Unless it is a UI element change RoleBinding to role binding through out the PR

Preeticp commented on modules/getting-started-openshift-gitops.adoc:
```diff
@@ -0,0 +1,128 @@
+  * If you only select *All Projects*, then you only need to provide the *Binding Type*.
+  * If the project is already selected as the one you want to create the RoleBinding for, the project does not ask for *Binding Type* because it assumes it is a RoleBinding.
+. Under *Role Binding*, provide a RoleBinding *Name* and *Namespace*.
+  * For the *Namespace*, specify the one you are granting permissions for.
```
remove bullet format

Preeticp commented on modules/getting-started-openshift-gitops.adoc:
```diff
@@ -0,0 +1,128 @@
+  * If the project is already selected as the one you want to create the RoleBinding for, the project does not ask for *Binding Type* because it assumes it is a RoleBinding.
+. Under *Role Binding*, provide a RoleBinding *Name* and *Namespace*.
+  * For the *Namespace*, specify the one you are granting permissions for.
+. For *Role*, select the *Cluster Role admin (CR admin)* from the dropdown list.
```
Select the *Cluster Role admin (CR admin)* from the *Role* list.

Preeticp commented on modules/getting-started-openshift-gitops.adoc:
```diff
@@ -0,0 +1,128 @@
+. Under *Subject*, select *Service Account* and provide the *Subject Namespace* and *Name*. For example:
+  * Subject Namespace: openshift-gitops
+  * Subject Name: argocd-cluster-argocd-application-controller
+. To create the RoleBinding, select *Create*.
```
Click *Create* to create the role binding.

Preeticp commented on modules/getting-started-openshift-gitops.adoc:
```diff
@@ -0,0 +1,128 @@
+. Go to the table of contents and make sure you are in the *Administrator* perspective and not the *Developer* perspective.
+. Update the existing *Subscription Object* for the GitOps operator.
+. Go the spec and add, *ARGOCD_CLUSTER_CONFIG_NAMESPACES*.
+. On the OpenShift Console, go to *Administration* > *CustomResourceDefinition* > *Subscription* > *Instances* then select *openshift-gitops-operator*.
```
Remove steps on line 81, 82, and 83. Line 84 is where the steps actually start describing what is on line 81, 82 and 83 accurately.

 In the *Administrator* perspective of the web Console, navigate to...

Preeticp commented on modules/getting-started-openshift-gitops.adoc:
```diff
@@ -0,0 +1,128 @@
+[id="setting-up-an-additional-permissions-for-argocd_{context}"]
+== Setting up additional permissions for ArgoCD
+
+If you want to expand the permissions granted to ArgoCD, create Cluster Roles with additional permissions and then a new ClusterRoleBinding to associate them to a Service Account.
```
cluster roles
...and then add a new cluster role binding...

Preeticp commented on modules/getting-started-openshift-gitops.adoc:
```diff
@@ -0,0 +1,128 @@
+
+If you want to expand the permissions granted to ArgoCD, create Cluster Roles with additional permissions and then a new ClusterRoleBinding to associate them to a Service Account.
+
+For an example, extend the permissions for the pre-installed ArgoCD instance to list the secrets for all namespaces. To extend these permissions, you must create a cluster role:
```
For example

Preeticp commented on modules/getting-started-openshift-gitops.adoc:
```diff
@@ -0,0 +1,128 @@
+For an example, extend the permissions for the pre-installed ArgoCD instance to list the secrets for all namespaces. To extend these permissions, you must create a cluster role:
+
+. Go to the table of contents and make sure you are in the *Administrator* perspective and not the *Developer* perspective.
+. Navigate to *User Management* > *Roles* > *Create Role*.
```
In the *Administrator* perspective of the web Console, navigate to...

Preeticp commented on modules/getting-started-openshift-gitops.adoc:
```diff
@@ -0,0 +1,128 @@
+
+. Go to the table of contents and make sure you are in the *Administrator* perspective and not the *Developer* perspective.
+. Navigate to *User Management* > *Roles* > *Create Role*.
+. Use the following ClusterRole Yaml template to add rules to specify the additional permissions, then select *Create*:
```
`ClusterRole` YAML template

Preeticp commented on modules/getting-started-openshift-gitops.adoc:
```diff
@@ -0,0 +1,128 @@
+. Select *Service Account* under *Subject* and provide the *Subject Namespace* and *Name*. For example:
+  * Subject Namespace: openshift-gitops
+  * Subject Name: argocd-cluster-argocd-application-controller
+. Select *Create* to create the ClusterRoleBinding.
```
cluster role binding

Preeticp commented on modules/getting-started-openshift-gitops.adoc:
```diff
@@ -0,0 +1,128 @@
+. Go to the table of contents and make sure you are in the *Administrator* perspective and not the *Developer* perspective.
+. Navigate to *User Management* > *Role Bindings* > *Create Binding*.
+. Select *Cluster-wide role binding (ClusterRoleBinding)* for the *Binding Type*.
+. Provide a unique RoleBinding name.
```
role binding

Preeticp commented on modules/getting-started-openshift-gitops.adoc:
```diff
@@ -0,0 +1,128 @@
+.Creating ClusterRoleBinding
+
+. Go to the table of contents and make sure you are in the *Administrator* perspective and not the *Developer* perspective.
+. Navigate to *User Management* > *Role Bindings* > *Create Binding*.
```
In the *Administrator* perspective of the web Console, navigate to... 
Arrows should use this format -> and not > in all instances in this PR

jc-berger commented on cicd/gitops/getting-started-with-openshift-gitops.adoc:
```diff
@@ -0,0 +1,19 @@
+[id="getting-started-with-openshift-gitops"]
+= Getting started with OpenShift GitOps
+include::modules/common-attributes.adoc[]
+//include::modules/gitops-document-attributes.adoc[]
```
I added the `//` before this include because this doc does not exist in this PR. This is the correct way to prevent build errors from occurring, right?

Preeticp replied:
We would want to use {gitops-title} in this assembly too, so I am not sure I understand why we want to comment this out.

jc-berger commented:
@Preeticp, new commit is up, addressing your feedback! I'll look into the build failure tomorrow...I suspect it's because of a doc I've deleted (outdated module). Thanks for the help. Now I'll await the dev SME review!

jc-berger commented on _topic_map.yml:
```diff
@@ -1277,16 +1277,14 @@ Topics:
-#  - Name: Getting started with OpenShift GitOps
-#    File: getting-satrted-with-openshift-gitops
+   - Name: Getting started with OpenShift GitOps
+     File: getting-satrted-with-openshift-gitops
```
typo on my part, hence, the build error. I know I had to look at it with a fresh pair of eyes to find what was wrong :) I'll correct this in my next commit

Preeticp commented on modules/setting-up-a-new-argocd-instance.adoc:
```diff
@@ -0,0 +1,85 @@
+. In the *Installed Operators* page, select *OpenShift GitOps* then the *Argo CD* tab.
+. Create an Argo CD instance and specify the following configurations:
+  * Name: <argocd-instance-name>
+  * Server > Route > Enable Route = true. This configuration creates an external OS route to access the Argo CD server.
```
Spell out first instance of an acronym. What does os stand for here?

jc-berger replied:
I said it means open source here. I would like that confirmed though!

Preeticp commented on modules/getting-started-with-cluster-configurations.adoc:
```diff
@@ -0,0 +1,39 @@
+// * openshift-docs/cicd/gitops/getting-started--with-openshift-gitops.adoc
+
+[id="getting-started-with-cluster-configuration_{context}"]
+= Getting started with cluster configuration
```
IMO this section should be titled something like: Launching the operator provided Argo CD instance Can you check with Shiubhik and Dewan if that works.
The current title is so generic, cluster configuration for what? does not give the end user any clue about what this section is about. And I am not sure what cluster configs we talk about here.

jc-berger replied:
Changed the name so folks can see how it looks before making up their minds :)

Preeticp commented on modules/setting-up-a-new-argocd-instance.adoc:
```diff
@@ -0,0 +1,85 @@
+// * openshift-docs/cicd/gitops/getting-started--with-openshift-gitops.adoc
+
+[id="setting-up-a-new-argocd-instance_{context}"]
+= Setting up a new Argo CD instance
```
How about: Setting up a customized Argo CD instance ?

Preeticp commented on modules/setting-up-a-new-argocd-instance.adoc:
```diff
@@ -0,0 +1,85 @@
+
+. In the *Administrator* perspective of the web console, navigate to *Operators* -> *Installed Operators*.
+. Create a new project or select an existing project, depending on where you want to install the Argo CD instance.
+. In the *Installed Operators* page, select *OpenShift GitOps* then the *Argo CD* tab.
```
and then the *Argo CD* tab.

Preeticp commented on modules/setting-up-a-new-argocd-instance.adoc:
```diff
@@ -0,0 +1,85 @@
+To create a role binding:
+
+. In the *Administrator* perspective of the web console, navigate to *User Management* -> *Role Bindings* > select *Create Binding*.
+  * You now see a form that guides you through the process of creating a role binding:
```
Remove the * make it part of the previous step.

Preeticp commented on modules/setting-up-a-new-argocd-instance.adoc:
```diff
@@ -0,0 +1,85 @@
+
+After you deploy an Argo CD instance, you can manage your cluster configurations:
+
+. In the *Administrator* perspective on the web console, navigate to *Administration* -> *CustomResourceDefinition* -> *Subscription* -> *Instances* then select *openshift-gitops-operator*.
```
and then select *openshift-gitops-operator*.

Preeticp commented on modules/setting-up-a-new-argocd-instance.adoc:
```diff
@@ -0,0 +1,85 @@
+After you deploy an Argo CD instance, you can manage your cluster configurations:
+
+. In the *Administrator* perspective on the web console, navigate to *Administration* -> *CustomResourceDefinition* -> *Subscription* -> *Instances* then select *openshift-gitops-operator*.
+. Select the *YAML* tab, edit the Subscription yaml, and add:
```
`Subscription` YAML file

Preeticp commented on modules/setting-up-a-new-argocd-instance.adoc:
````diff
@@ -0,0 +1,85 @@
+```
+
+[id="setting-up-an-additional-permissions-for-argocd_{context}"]
+== Setting up additional permissions for Argo CD
````
This is the only section that uses second level heading with ID. Either use the `.Title` or the second level heading `== Title` consistently.

Preeticp commented on modules/setting-up-a-new-argocd-instance.adoc:
```diff
@@ -0,0 +1,85 @@
+
+If you want to expand the permissions granted to Argo CD, create cluster roles with additional permissions and then add a new cluster role binding to associate them to a service account.
+
+For example, extend the permissions for the pre-installed Argo CD instance to list the secrets for all namespaces. To extend these permissions, you must create a cluster role:
```
Modify this section to comprise 2 main steps: 1. create a cluster role 2. add a new cluster role binding to associate them to a service account.
Suggest you start it like this:
s/ For example, extend the permissions for the pre-installed Argo CD instance to list the secrets for all namespaces./For example, to extend the permissions for the pre-installed Argo CD instance so that it can list the secrets for all namespaces:
.  Create a cluster role:

Preeticp commented on modules/setting-up-a-new-argocd-instance.adoc:
```diff
@@ -0,0 +1,85 @@
+
+For example, extend the permissions for the pre-installed Argo CD instance to list the secrets for all namespaces. To extend these permissions, you must create a cluster role:
+
+. In the *Administrator* perspective of the web console, navigate to *User Management* -> *Roles* -> *Create Role*.
+. Use the following `ClusterRole` Yaml template to add rules to specify the additional permissions, then select *Create*:
```
make both these steps nested steps, use `..` instead of `.`

Preeticp commented on modules/setting-up-a-new-argocd-instance.adoc:
```diff
@@ -0,0 +1,85 @@
+
+. In the *Administrator* perspective of the web console, navigate to *User Management* -> *Roles* -> *Create Role*.
+. Use the following `ClusterRole` Yaml template to add rules to specify the additional permissions, then select *Create*:
+
```
You will need to add a + sign above the code snippet to align it with the prev. step.

Preeticp commented on modules/setting-up-a-new-argocd-instance.adoc:
````diff
@@ -0,0 +1,85 @@
+  verbs: ["get", "watch", "list"]
+```
+
+.Creating cluster role binding
````
This would be the second main step in this section:
. Create a cluster role binding:

Preeticp commented on modules/setting-up-a-new-argocd-instance.adoc:
```diff
@@ -0,0 +1,85 @@
+
+.Creating cluster role binding
+
+. In the *Administrator* perspective of the web console, navigate to *User Management* -> *Role Bindings* -> *Create Binding*.
+. Select *Cluster-wide role binding (ClusterRoleBinding)* for the *Binding Type*.
+. Provide a unique role binding name.
+. Select the new cluster role or an existing cluster role.
+. Select *Service Account* under *Subject* and provide the *Subject Namespace* and *Name*. For example:
+  * Subject Namespace: openshift-gitops
+  * Subject Name: argocd-cluster-argocd-application-controller
+. Select *Create* to create the cluster role binding.
```
All these would become nested steps, use ..
Lines 83 and 84 would need ***, just check in the preview that it works, if not try **

Preeticp commented on modules/setting-up-an-openshift-login.adoc:
```diff
@@ -0,0 +1,90 @@
+
+To set up an OpenShift login, you must create a new client in Keycloak:
+
+. Log into your Keycloak server.
```
in to your

Preeticp commented on modules/setting-up-an-openshift-login.adoc:
```diff
@@ -0,0 +1,90 @@
+To set up an OpenShift login, you must create a new client in Keycloak:
+
+. Log into your Keycloak server.
+. Navigate to *myrealm* -> *Clients* -> then select *Create*:
```
and then select

Preeticp commented on modules/setting-up-an-openshift-login.adoc:
```diff
@@ -0,0 +1,90 @@
+
+. Log into your Keycloak server.
+. Navigate to *myrealm* -> *Clients* -> then select *Create*:
+  * For *Client ID*, select *argocd*.
+  * For *Client Protocol*, select *openid-connect*.
+  * For *Root URL*, select the Argo CD route URL.
```
All these should use ..  for nested steps and not *

Preeticp commented on modules/setting-up-an-openshift-login.adoc:
```diff
@@ -0,0 +1,90 @@
+  * For *Client Protocol*, select *openid-connect*.
+  * For *Root URL*, select the Argo CD route URL.
+. Configure the client:
+  * For *Access Type*, select *confidential*.
+  * For *Valid Redirect URLs*, select the callback url for your Argo CD hostname. Your callback should be `https://{hostname}/auth/callback`.
+  * Set the *Base URL* to */applications*.
+  * Save your work and then you get the *Credentials* tab. Go to this tab.
+  * For *Client Authenticator*, select *Client ID and Secret* and copy the *argocd-secret* so you can use it for your Argo CD configuration.
```
All these should use .. for nested steps and not *

Preeticp commented on modules/setting-up-an-openshift-login.adoc:
```diff
@@ -0,0 +1,90 @@
+  * Save your work and then you get the *Credentials* tab. Go to this tab.
+  * For *Client Authenticator*, select *Client ID and Secret* and copy the *argocd-secret* so you can use it for your Argo CD configuration.
+
+.Configuring Argo CD OIDC
```
Please expand OIDC the first time you use it.

jc-berger replied:
I said it means OpenID Connect...not sure if this is correct though!

Preeticp replied:
@sbose78 can you confirm, please?

sbose78 replied:
That's correct

Preeticp commented on modules/setting-up-an-openshift-login.adoc:
```diff
@@ -0,0 +1,90 @@
+
+After you create a new client in Keycloak and generate an *argocd-secret* in the Argo CD secret, you get a client secret. To configure Argo CD OIDC, store this client secret with the following instructions:
+
+. Encode the client secret in base64 by running:
```
remove 'by running' in all the steps.

Preeticp commented on modules/setting-up-an-openshift-login.adoc:
```diff
@@ -0,0 +1,90 @@
+  oidc.keycloak.clientSecret: ODMwODM5NTgtOGVjNi00N2IwLWE0MTEtYThjNTUzODFmYmQy
+----
+
+To enable your Keycloak authentication, add the OIDC configuration and edit the Argo CD CR by running:
```
Add . before the sentence, it is another step.

Preeticp commented on modules/setting-up-an-openshift-login.adoc:
```diff
@@ -0,0 +1,90 @@
+      enabled: true
+----
+
+Verify the following:
```
add . this is another step.

sbose78 commented on modules/setting-up-an-openshift-login.adoc:
```diff
@@ -0,0 +1,90 @@
+      - TaskRun
+      - PipelineRun
+  oidcConfig: |
+    name: OpenShift Single Sign-OnKeycloak
```
"OpenShift Single Sign-On"

sbose78 commented on modules/setting-up-an-openshift-login.adoc:
```diff
@@ -0,0 +1,90 @@
+
+. Log into your Keycloak server.
+. Navigate to *myrealm* -> *Clients* -> then select *Create*:
+  * For *Client ID*, select *argocd*.
```
/s/select/type

sbose78 commented on modules/setting-up-a-new-argocd-instance.adoc:
```diff
@@ -0,0 +1,85 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  # "namespace" omitted since ClusterRoles are not namespaced
```
>  "namespace" omitted since ClusterRoles are not namespaced

comment is not needed

jc-berger commented:
@Preeticp and @sbose78, thank you both for the awesome review :) 

The big issue I have is that I see this branch has conflicts and I am unable to rebase my branch because of the conflicting _topic_map.yml. Either of you know what could be causing this issue?

Preeticp commented on cicd/gitops/getting-started-with-openshift-gitops.adoc:
```diff
@@ -0,0 +1,20 @@
+
+== Prerequisites
+
+* You must install the xref:../../operators/admin/olm-adding-operators-to-cluster.adoc#olm-adding-operators-to-a-cluster[Red Hat OpenShift GitOps operator] listed in the OpenShift OperatorHub. Go with the default installation option. Once installed, the operator is applicable to all namespaces on the entire cluster. When you install the Red Hat OpenShift GitOps operator, you get an instance of Ar ...
```
S/Go with the default installation option./Follow the default....

Preeticp commented on modules/launching-the-operator-provided-argocd-instance.adoc:
```diff
@@ -0,0 +1,31 @@
+
+When you install Red Hat OpenShift GitOps, the operator automatically sets up a ready-to-use Argo CD instance for cluster configuration. Argo CD manages specific cluster-scoped resources, including: platform operators, optional Operator Lifecycle Manager (OLM) operators, and user management.
+
+To use Argo CD, you must launch into the Argo CD instance from the Console Application Launcher:
```
If this is the name in the GUI, make it bold.

Preeticp commented on modules/setting-up-a-customized-argocd-instance.adoc:
```diff
@@ -0,0 +1,80 @@
+
+To create a role binding:
+
+. In the *Administrator* perspective of the web console, navigate to *User Management* -> *Role Bindings* > select *Create Binding*. You now see a form that guides you through the process of creating a role binding:
```
select *Create Binding* either remove select from this or remove the arrow and add 'and then select..'
arrow needs to be -> in case you retain it.

Preeticp commented on modules/setting-up-a-customized-argocd-instance.adoc:
```diff
@@ -0,0 +1,80 @@
+. In the *Administrator* perspective of the web console, navigate to *User Management* -> *Role Bindings* > select *Create Binding*. You now see a form that guides you through the process of creating a role binding:
+. For *Binding Type*, select *Namespace Role Binding (RoleBinding)*. You do not need to select the binding type if a particular project is selected, as the type is assumed to be role binding for the project.
+. Under *Role Binding*, provide a role binding *Name* and *Namespace*. For the *Namespace*, specify the one you are granting permissions for.
+. Select the *Cluster Role admin (CR admin)* from the *Role* list.
```
In the *Role* list, select... 
First location, then action

Preeticp commented on modules/setting-up-a-customized-argocd-instance.adoc:
```diff
@@ -0,0 +1,80 @@
+
+After you deploy an Argo CD instance, you can manage your cluster configurations:
+
+. In the *Administrator* perspective on the web console, navigate to *Administration* -> *CustomResourceDefinition* -> *Subscription* -> *Instances* and then select *openshift-gitops-operator*.
```
comma after *Instances*

Preeticp commented on modules/setting-up-an-openshift-login.adoc:
```diff
@@ -0,0 +1,90 @@
+----
+$ kubectl edit secret argocd-secret -n argocd
+----
+*** Your secret now looks like the following:
```
Remove ***

Preeticp commented on modules/setting-up-an-openshift-login.adoc:
```diff
@@ -0,0 +1,90 @@
+----
+$ kubectl edit argocd -n argocd
+----
+*** Your Argo CD CR now looks like the following:
```
Remove ***

Preeticp commented on modules/setting-up-a-customized-argocd-instance.adoc:
```diff
@@ -0,0 +1,80 @@
+. In the *Installed Operators* page, select *OpenShift GitOps* and then the *Argo CD* tab.
+. Create an Argo CD instance and specify the following configurations:
+*** Name: <argocd-instance-name>
+*** Server > Route > Enable Route = true. This configuration creates an external open source (OS) route to access the Argo CD server.
```
Server > Route > Enable Route: enclose each of these in asterix, making them bold and use ->

Preeticp commented on modules/setting-up-a-customized-argocd-instance.adoc:
```diff
@@ -0,0 +1,80 @@
+
+If you want to install your own instance for managing your cluster configurations or deploying your applications, you must deploy your own instance of Argo CD. Using your own instance of Argo CD helps you to continuously deliver your applications.
+
+.Installing ArgoCD
```
@jc-berger PTAL at the preview: https://deploy-preview-29411--osdocs.netlify.app/openshift-enterprise/latest/cicd/gitops/getting-started-with-openshift-gitops.html
these sections are rendered in red and italics. I suggest we change the headings from . to == second level headings everywhere.
Please use th epreview to do a self edit, before you send it out to the OCP team.

sbose78 commented:
Looks good to me!

kalexand-rh commented on modules/setting-up-an-openshift-login.adoc:
```diff
@@ -0,0 +1,90 @@
++
+[source,terminal]
+----
+$ echo -n '83083958-8ec6-47b0-a411-a8c55381fbd2' | base64
```
```suggestion
$ echo -n '<argocd-secret>' | base64 <1>
```

Then after the codeblock, add something like:

```
<1> Enter the `<argocd-secret>` value that you obtained from your Keycloak server.
```

kalexand-rh commented on modules/setting-up-an-openshift-login.adoc:
```diff
@@ -0,0 +1,90 @@
+$ echo -n '83083958-8ec6-47b0-a411-a8c55381fbd2' | base64
+----
+
+. Add the base64 value to a new key called `oidc.keycloak.clientSecret` and edit the secret:
```
From this point on, you've completely lost me. The descriptions of the steps don't appear to correspond to the commands that you're using, and I am extremely confused what of the YAML you're showing as example _is actually there_ and what the user needs to customize. The YAML format itself looks slightly off - I've never seen a YAML example start with a "$." Please review the rest of the module and try to be clearer. Use the annotation format from some of the other YAML examples to explain what's going on.

kalexand-rh commented on modules/setting-up-an-openshift-login.adoc:
```diff
@@ -0,0 +1,90 @@
+    route:
+      enabled: true
+----
+. Verify the following:
```
I would not make this a separate step. I'd make these checks annotations on the YAML itself. This format is hard to follow.

kalexand-rh reviewed (commented):
This looks like two different user stories wedged into the same assembly, you have some malformed modules, and I am confused about part of the flow.

iam-veeramalla commented on cicd/gitops/getting-started-with-openshift-gitops.adoc:
```diff
@@ -0,0 +1,20 @@
+//Procedure Module
+include::modules/setting-up-a-customized-argocd-instance.adoc[leveloffset=+1]
+
+//Procedure Module
```
Keycloak procedure is common irrespective of configuring it with the ArgoCD instance provided by Operator in openshift-gitops namespace or any other instance created by the user in a different namespace.

sbose78 commented on modules/setting-up-a-customized-argocd-instance.adoc:
```diff
@@ -0,0 +1,80 @@
+  config:
+    env:
+    - name: ARGOCD_CLUSTER_CONFIG_NAMESPACES
+      Value: openshift-gitops, <namespace where argocd instance is installed>
```
`openshift-gitops` needs to be there. We are essentially appending to an existing env var.

sbose78 commented on modules/setting-up-a-customized-argocd-instance.adoc:
```diff
@@ -0,0 +1,80 @@
+
+== Setting up additional permissions for Argo CD
+
+To extend the permissions for the pre-installed Argo CD instance so that it can list the secrets for all namespaces:
```
> To extend the permissions for the pre-installed Argo CD instance so that it can list the secrets for all namespaces:

Could we indicate that this is only an example and the admin needs to _handle with care_

Preeticp commented:
Hi @kalexand-rh thanks a ton for your thorough review, appreciate it. 
Post discussions with the dev team, we thought it best to add in a brief getting started for the TP release, and then add in the other changes incrementally. 
We have removed the: modules/setting-up-an-openshift-login.adoc and the modules/setting-up-a-customized-argocd-instance.adoc from this PR, and have created Jiras to fix them post GA: https://issues.redhat.com/browse/RHDEVDOCS-2695 and https://issues.redhat.com/browse/RHDEVDOCS-2696. The Jiras have links to the relevant comment for both modules so that we have a record and fix all the comments.

dewan-ahmed commented:
Hi, few comments:

1. On step 3d. `Select the argo-cluster-cluster instance to display the password.`, I think the name of the password is `argocd-cluster-cluster`
![access-secret-1](https://user-images.githubusercontent.com/30604461/108752355-bbc7d580-7519-11eb-8a9d-8bdb5dabe727.png)
2. From what I read, the getting started doc is guiding the customer up to the running ArgoCD instance. Was that the entire scope of the getting started doc? My take on this is providing more information to a customer i.e. how to make use of that instance in terms of creating a sample argocd application. Since this is a new product and not so much content exist on this online, these docs would be the only go-to places so my preference is to provide a bit more info. @sbose78 @wtam2018 can comment further.

sbose78 commented:
Getting started definitely needs to involve

* Installing OpenShift GitOps/ArgoCD as an "admin".
* Enabling users/admins to deploy applications with it. 

We don't need all of it to go in together though :)

sbose78 reviewed (approved):
👍

kalexand-rh reviewed (commented):
I think that this looks good and has a much better scope for a "Getting started" topic!

codyhoag commented:
/cherrypick enterprise-4.7
